ios/brave-ios/Sources/Brave/Frontend/Browser/TabManager.swift

```diff
@@ -481,6 +483,9 @@ class TabManager: NSObject {
   ) -> any TabState {
     let tab = addTab(request, afterTab: afterTab, isPrivate: isPrivate)
     selectTab(tab)
+    if tab.visibleURL == TabManager.ntpInteralURL {
+      delegates.forEach { $0.get()?.tabManagerDidAddAndSelectNTP(self) }
+    }
```

Comment:
This seems like an odd solution to me. Why is this part of the TabManager logic which is just meant for tab storage?  It seems like either we'd want to add additional actions where we call `addTab` knowing that we're opening a blank new tab anyways. Is it that we just dont have a centralized place where that happens?

If so then why dont we do it somewhere like where NTP view controller is created for a new tab?

Reply:
moved the logic to where we create and show NTP